hack/ci/e2e.sh

```diff
@@ -93,17 +93,50 @@ build() {
 
 # up a cluster with kind
 create_cluster() {
+    # create the config file
+    cat <<EOF > "${ARTIFACTS}/kind-config.yaml"
+# config for 1 control plane node and 2 workers
+# necessary for conformance
+kind: Config
+apiVersion: kind.sigs.k8s.io/v1alpha2
+nodes:
+# the control plane node
+- role: control-plane
+- role: worker
+  replicas: 2
+EOF
     # mark the cluster as up for cleanup
     # even if kind create fails, kind delete can clean up after it
     KIND_IS_UP=true
-    kind create cluster --image="kindest/node:latest" --retain --wait=1m --loglevel=debug
+    # actually create, with:
+    # - do not delete created nodes from a failed cluster create (for debugging)
+    # - wait up to one minute for the nodes to be "READY"
+    # - set log leve to debug
+    # - use our multi node config
+    kind create cluster \
+        --image=kindest/node:latest \
+        --retain \
+        --wait=1m \
+        --loglevel=debug \
+        "--config=${ARTIFACTS}/kind-config.yaml"
 }
 
 # run e2es with kubetest
 run_tests() {
+    # export the KUBECONFIG
+    KUBECONFIG="$(kind get kubeconfig-path)"
+    export KUBECONFIG
+
     # base kubetest args
     KUBETEST_ARGS="--provider=skeleton --test --check-version-skew=false"
 
+    # get the number of worker nodes
+    # TODO(bentheelder): this is kinda gross
+    NUM_NODES="$(kubectl get nodes \
+        -o=jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.spec.taints}{"\n"}{end}' \
+        | grep -cv "node-role.kubernetes.io/master" \
+    )"
```

Comment:
do we want to add a helper command for that in kind?
given kubectl already prints detailed information, we might simply add output for how many CP and worker nodes there are:

e.g
```
kind get num-nodes control-plane // 1
kind get num-nodes worker // 2
kind get num-nodes // 3
```

as per our "what is a node in kind" discussion this should exclude external LBs.

Reply:
Possibly. An alternative options is that e2e.test could have fallback detection of this (I poked around and I don't think it does).. if it doesn't have that, this will be a problem for sonobuoy I think.

This didnt take long and should work pretty fine so I didn't dig further yet.